lib/block_view_dao_coin_limit_order.go

```diff
@@ -37,37 +38,135 @@ func (bav *UtxoView) _connectDAOCoinLimitOrder(
+	transactorPKID := bav.GetPKIDForPublicKey(txn.PublicKey).PKID
+
+	if transactorPKID == nil {
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderInvalidTransactorPKID
+	}
+
+	// Validate DenominatedCoinType is one of our supported enum values and is always $DESO for now.
+	switch txMeta.DenominatedCoinType {
+	case DAOCoinLimitOrderEntryDenominatedCoinTypeDESO:
+		break
+	case DAOCoinLimitOrderEntryDenominatedCoinTypeDAOCoin:
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderUnsupportedDenominatedCoinType
+	default:
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderUnsupportedDenominatedCoinType
+	}
+
+	// If denominated in $DESO, validate DenominatedCoinCreatorPKID is all zeroes.
+	if txMeta.DenominatedCoinType == DAOCoinLimitOrderEntryDenominatedCoinTypeDESO {
+		if !reflect.DeepEqual(ZeroPKID, txMeta.DenominatedCoinCreatorPKID) {
+			return 0, 0, nil, RuleErrorDAOCoinLimitOrderInvalidDAOCoinCreatorPKID
+		}
+	}
+
+	// If denominated in a DAO coin, validate DenominatedCoinCreatorPKID exists and has a profile.
+	if txMeta.DenominatedCoinType == DAOCoinLimitOrderEntryDenominatedCoinTypeDAOCoin {
+		// We currently don't support DAO coins as the denominated type.
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderUnsupportedDenominatedCoinType
+	}
+
+	// Validate DAOCoinCreatorPKID exists and has a profile.
+	// TODO
+
+	// Validate OperationType is one of our supported enum values.
+	switch txMeta.OperationType {
+	case DAOCoinLimitOrderEntryOrderTypeAsk:
+		break
+	case DAOCoinLimitOrderEntryOrderTypeBid:
+		break
+	default:
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderUnsupportedOperationType
+	}
+
+	// Validate price > 0.
+	if !txMeta.PriceNanos.Gt(uint256.NewInt()) {
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderInvalidPrice
+	}
+
+	// If denominated in $DESO, confirm PriceNanos is uint64.
+	if !txMeta.PriceNanos.IsUint64() {
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderInvalidPrice
+	}
+
+	// If denominated in DAO coins, confirm PriceNanos is uint256.
 	// TODO
 
-	// ----- Begin validations
-	// Create rule errors for each of these validations below
-	// lib/errors.go
-
-	// Validate TransactorPKID exists
-	// Validate DenominatedCoinType is one of our supported enum values and is DESO for now
-	// Validate DenominatedCoinCreatorPKID exists and has a profile or is all zeroes if $DESO
-	// Validate DAOCoinCreatorPKID exists and has a profile
-	// Validate OperationType is one of our supported enum values
-	// Validate PriceNanos > 0
-	//   If denominated in DESO, confirm PriceNanos is uint64
-	//   PriceNanos == uint256.ToUint64(PriceNanos).ToUint256()
-	// Validate Quantity > 0
-	// If DESO buy, validate quantity < max uint64, order.Quantity.IsUint64()
-	// If DESO buy, validate price * quantity < max uint64
+	// Validate quantity > 0.
+	if !txMeta.Quantity.Gt(uint256.NewInt()) {
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderInvalidQuantity
+	}
+
+	// Validate quantity is uint64.
+	if !txMeta.Quantity.IsUint64() {
```

Comment:
```suggestion
	if txMeta.DenominatedCoinType == DAOCoinLimitOrderEntryDenominatedCoinTypeDESO && !txMeta.Quantity.IsUint64() {
```

Reply:
I think we only want it to be less than max uint64 if it's denominated in DESO. Although this makes me a little concerned now since I expect DAO coin to be valued far less than DESO.  let's chat about this offline.

Reply:
Price refers to $DESO or DAO coin denomination.
Quantity refers solely to the number of DAO coins for sale.
I'm not sure we have to require that Quantity is a uint64.
It would just have to be less than or equal to the max supply of DAO coins.
What is the global max of DAO coins? Or is there a way to check the max number of DAO coins by PKID?